src/events/neo4j.js

```diff
@@ -0,0 +1,387 @@
+                    dbApiEvent,
+                    { items_count: records.length, operation_executed: 'Read' }
+                );
+                break;
+
+            case 'w':
+                eventInterface.addToMetadata(
+                    dbApiEvent,
+                    { operation_executed: 'Write' },
+                    { write_stats: summary.counters._stats }
+                );
+                break;
+            case 'rw':
+                eventInterface.addToMetadata(
+                    dbApiEvent,
+                    { items_count: records.length, operation_executed: 'Read, Write' },
+                    { write_stats: summary.counters._stats }
+                );
+                break;
+
+            default:
+                utils.debugLog(`Unkown query type: ${summary.queryType}`);
+                break;
+            }
+
+            dbApiEvent.setDuration(utils.createDurationTimestamp(startTime));
+            tracer.addEvent(dbApiEvent);
+            return observer.onCompleted.apply(session, sessionArguments);
+        },
+
+        // eslint-disable-next-line require-jsdoc
+        onError(err) {
+            eventInterface.setException(dbApiEvent, err);
+            // eslint-disable-next-line prefer-rest-params
+            return observer.onError.apply(session, sessionArguments);
+        },
+    };
+}
+
+
+/**
+ * Create new serverlessEvent for Neo4J run event (any type of run) and
+ *  add relevant metadata
+ * @param {neo4j.Session|neo4j.Transaction} session A DB session (could be Session of Transaction)
+ * @param {number} startTime The Event start time
+ * @returns {serverlessEvent.Event} The new Event
+ */
+function createNewNeo4jEvent(session, startTime) {
+    const { host, port } = getAddressInfo(session);
+
+    const {
+        metadata, fullMetadata,
+    } = getNeo4JSessionMetadata(session);
+
+    metadata.port = port;
+
+    const resource = new serverlessEvent.Resource([
+        host,
+        'neo4j',
+        metadata.operation,
+    ]);
+
+    const dbApiEvent = new serverlessEvent.Event([
+        `neo4j-${uuid4()}`,
+        utils.createTimestampFromTime(startTime),
+        null,
+        'neo4j',
+        0,
+        errorCode.ErrorCode.OK,
+    ]);
+
+    dbApiEvent.setResource(resource);
+
+    eventInterface.addToMetadata(dbApiEvent, metadata, fullMetadata);
+
+    return dbApiEvent;
+}
+
+
+/**
+ * Wraps neo4j.Session.run functions with tracing
+ * @param {Function} wrappedFunction The function to wrap from Neo4j
+ * @returns {Function} The wrapped function
+ */
+function neo4jSessionRunWrapper(wrappedFunction) {
+    return function internalNeo4jSessionRunWrapper(...args) {
+        const relevantArgs = getArgsFromFunction(...args, this.constructor.name);
+        const {
+            query, params, transactionConfig,
+        } = relevantArgs;
+
+        utils.debugLog('User called Neo4j wrapped Session run function');
+
+        const resultResponse = wrappedFunction.apply(
```

Comment:
Now if the customer's code throws an exception we will not add it to the tracer object

Reply:
fixed